docs/scos/dev/upgradability-services/upgradability-reference.md

```diff
@@ -110,3 +110,201 @@ The method name should be unique to the extent of making it impossible to accide
 2. Override the core method with the custom one you've created in the previous step.
 
 When the core method is overriden with a custom one, re-evaluate the code. The same error shouldn't be returned.
+
+
+
+## Non-public API class was extended or used
+
+Private API updates can break backward compatibility. So, backward compatibility in minor releases is not guaranteed in the private API. For example, if you use a core class or method on the project level, and it is updated or removed, the error may occur during an update.
+
+Also there are exceptions, models on project level can extends from Core:
+
+1. Facade, Factory, Entity manager, Repository, Dependency provider, Config, ConfigurationProvider.
+
+2. All models from Kernel, Bootstrap, and Development modules.
+
+### Overriding a core method on the project level
+
+To avoid the error during updates and achieve the same result, provide the custom business model in the private API.
+
+### Example of code that can cause the error and example of the error itself
+
+`CustomerAccessForm` extends from `Spryker\Zed\CustomerAccessGui\Communication\Form\CustomerAccessForm` class from the core level.
+
+```php
+<?php
+...
+**
+ * @method \Spryker\Zed\CustomerAccessGui\Communication\CustomerAccessGuiCommunicationFactory getFactory()
+ */
+class CustomerAccessForm extends SprykerCustomerAccessForm
+{
+    public const OPTION_CONTENT_TYPE_ACCESS_MANAGEABLE = 'OPTION_CONTENT_TYPE_ACCESS_MANAGEABLE';
+    public const OPTION_CONTENT_TYPE_ACCESS_NON_MANAGEABLE = 'OPTION_CONTENT_TYPE_ACCESS_NON_MANAGEABLE';
+    public const OPTION_CONTENT_TYPE_ACCESS_NON_MANAGEABLE_DATA = 'OPTION_CONTENT_TYPE_ACCESS_NON_MANAGEABLE_DATA';
+...
+}
+```
+
+Related error in the Evaluator output:
+```text
+------------------------------------------------------------------------------------------------------------------------
+Pyz\Zed\CustomerAccessGui\Communication\Form\CustomerAccessForm
+"Please avoid dependency: Spryker\\Zed\\CustomerAccessGui\\Communication\\Form\\CustomerAccessForm in Pyz\\Zed\\CustomerAccessGui\\Communication\\Form\\CustomerAccessForm"
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### How do I achieve the same result without the error?
+
+To resolve the error provided in the example, do the following:
+
+1. Change extending from `Spryker\Zed\CustomerAccessGui\Communication\Form\CustomerAccessForm` to extending from `Spryker\Zed\Kernel\Communication\Form\AbstractType`
```

Comment:
How is `Spryker\Zed\Kernel\Communication\Form\AbstractType` related to `Spryker\Zed\CustomerAccessGui\Communication\Form\CustomerAccessForm`?

Reply:
`Spryker\Zed\CustomerAccessGui\Communication\Form\CustomerAccessForm` extends from `Spryker\Zed\Kernel\Communication\Form\AbstractType`. 

We cant use (extends) `CustomerAccessForm` on the Project level but can do it with `AbstractType` because it is one of the exceptions.